systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthPlainST.java

```diff
@@ -0,0 +1,293 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.greaterThan;
+
+public class OauthPlainST extends OauthST {
+
+    @Test
+    void testProducerConsumer() {
+        testMethodResources().producerWithOauth(oauthTokenEndpointUri, TOPIC_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME)).done();
+        testMethodResources().consumerWithOauth(null, oauthTokenEndpointUri, TOPIC_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME)).done();
+
+        String producerPodName = kubeClient().listPodsByPrefixInName("hello-world-producer-").get(0).getMetadata().getName();
+        String producerMessage = "Sending messages \"Hello world - 9\"";
+
+        StUtils.waitUntilMessageIsInPodLogs(producerPodName, producerMessage);
+
+        String producerLogs = kubeClient().logs(producerPodName);
+        int messageCount = 10;
+
+        for (int i = 0; i < messageCount; i++) {
+            assertThat("Producer doesn't send message" + i, producerLogs, containsString("Sending messages \"Hello world - " + i + "\""));
+        }
+
+        String consumerPodName = kubeClient().listPodsByPrefixInName("hello-world-consumer-").get(0).getMetadata().getName();
+        String consumerMessage = "value: \"Hello world - 9\"";
+
+        StUtils.waitUntilMessageIsInPodLogs(consumerPodName, consumerMessage);
+
+        String consumerLogs = kubeClient().logs(consumerPodName);
+
+        for (int i = 0; i < messageCount; i++) {
+            assertThat("Producer doesn't send message" + i, consumerLogs, containsString("value: \"Hello world - " + i + "\""));
+        }
+    }
+
+    @Test
+    void testProducerConsumerStreams() {
+        testMethodResources().producerWithOauth(oauthTokenEndpointUri, TOPIC_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME)).done();
+        testMethodResources().consumerWithOauth(null, oauthTokenEndpointUri, "my-topic-reversed", KafkaResources.plainBootstrapAddress(CLUSTER_NAME)).done();
+        testMethodResources().kafkaStreamsWithOauth(oauthTokenEndpointUri, KafkaResources.plainBootstrapAddress(CLUSTER_NAME)).done();
+
+        String producerPodName = kubeClient().listPodsByPrefixInName("hello-world-producer-").get(0).getMetadata().getName();
+        String producerMessage = "Sending messages \"Hello world - 10\"";
+
+        StUtils.waitUntilMessageIsInPodLogs(producerPodName, producerMessage);
+
+        String producerLogs = kubeClient().logs(producerPodName);
+        int messageCount = 10;
+
+        for (int i = 0; i < messageCount; i++) {
+            assertThat("Producer doesn't send message" + i, producerLogs, containsString("Sending messages \"Hello world - " + i + "\""));
+        }
+
+        String consumerPodName = kubeClient().listPodsByPrefixInName("hello-world-consumer-").get(0).getMetadata().getName();
+        String consumerMessage = "value: \"9 - dlrow olleH\"";
```

Comment:
why as other test don't wait for latest "10" indexed message but "9" instead here?

Reply:
When i will want to test 10 i will need to add if clause inside my loop because message will like this -> `"value: \"01 - dlrow olleH\"";` and not "value: \"10 - dlrow olleH\"";`